File: lib/models/tv_channel.dart
class TvChannel {
  const TvChannel({required this.id, required this.image, required this.name, required this.resolutions});

  final String id;
  final String image;
  final String name;
  final Map<String, String> resolutions;

  Map<String, dynamic> toJson() => {'id': id, 'name': name, 'image': image, 'resolutions': resolutions};

  @override
  String toString() => 'TvChannel{id: $id, image: $image, name: $name}';
}

File: android/src/main/kotlin/uz/shs/video_player/models/TvChannel.kt
package uz.shs.video_player.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class TvChannel(
    @SerializedName("id")
    val id: String,
    @SerializedName("image")
    val image: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("resolutions")
    var resolutions: HashMap<String, String>
) : Serializable